### ISSUE TITLE
Update documentation comment for invalidationBehavior

### ISSUE THREAD

**Issue by cainePW** [reduxjs/redux-toolkit#4587]
It seems the default value `invalidationBehavior` does not accurately reflect the documentation comment.

The [comment](https://github.com/reduxjs/redux-toolkit/blob/75b4f698e1c58efa6b19096abe5850da0a431da0/packages/toolkit/src/query/createApi.ts#L163) describes the default value as `immediately`.

Looking at the default set of options [here](https://github.com/reduxjs/redux-toolkit/blob/75b4f698e1c58efa6b19096abe5850da0a431da0/packages/toolkit/src/query/createApi.ts#L262)  it seems the default value is set to `delayed`.

```js
const optionsWithDefaults: CreateApiOptions<any, any, any, any> = {
      reducerPath: 'api',
      keepUnusedDataFor: 60,
      refetchOnMountOrArgChange: false,
      refetchOnFocus: false,
      refetchOnReconnect: false,
      invalidationBehavior: 'delayed',
      ...options,
      extractRehydrationInfo,

...
```

If I understand this correctly, the comment needs to be updated. Apologies if I've got this wrong!

Thanks for your hard work on this library!

**Comment by Kluskey** [comment]
+1. This can cause issues with listeners that need to immediately respond to fulfilled queries. I had a scenario where it created an odd discrepancy where an API request succeeded but in the listener logic, the state wasn't up to date because the cache tags hadn't been invalidated (_yet_, because they were `delayed`).

**Comment by EskiMojo14** [comment]
Good spot - PRs welcome :)

**Comment by markerikson** [comment]
Looks like this was fixed in #4555
